### PR TITLE
fix: Fix linear gradient border not working on Wasm

### LINKED ISSUE #6751: Support for `LinearGradientBrush` on borders on WASM
## What would you like to be added:

Support for LinearGradientBrush on borders on WASM

## Why is this needed:

WinUI 2.6

## For which Platform:

- [ ] iOS
- [ ] Android
- [x] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] macOS
- [ ] Skia
  - [ ] WPF
  - [ ] GTK (Linux)
  - [ ] Tizen
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

## Anything else we need to know?

CSS `attr` and `var` may be useful for this

### PR DESCRIPTION
**Pull request by Youssef1313**
Hide whitespace changes for easier review of the test

GitHub Issue (If applicable): closes #

## PR Type

What kind of change does this PR introduce?

- Bugfix

## What is the current behavior?

Linear gradient border not working

## What is the new behavior?

Working:

image

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/BorderTests/Border_LinearGradientBrush.xaml [hunk 1]
```diff
@@ -9,40 +9,52 @@
...
+					Width="150"
+					Height="100"
+					CornerRadius="40"
+					BorderThickness="4"
+					BorderBrush="Black">
+				<Border.Background>
+					<LinearGradientBrush StartPoint="0,0"
+										 EndPoint="1,1">
+						<GradientStop Color="Yellow"
+									  Offset="0.0" />
```
Comment (Youssef1313) [inline src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/BorderTests/Border_LinearGradientBrush.xaml:34]: This existing test was already working. The new test case was broken and is now fixed.

**Comment by Youssef1313** [comment]
@jeromelaban @MartinZikmund I'm surprised that CI didn't report screenshot differences.

**Comment by MartinZikmund** [comment]
@Youssef1313 I think the problem with the currently supported version of gradient on border is that it does not work with rounded corners. Can you try to see what happens when you add some `CornerRadius`?

**Comment by Youssef1313** [comment]
@MartinZikmund Indeed the radius doesn't seem to be working correctly:

radius

**Comment by Youssef1313** [comment]
I'll remove the "closes" keyword part to keep the issue open since I couldn't yet figure out how to make the border work as expected in CSS. Tried various solutions I found on stackoverflow but none worked for me.

**Review by carldebilly — APPROVED** [review]
That's amazing!  Good work!  I didn't know it was possible to use `background-image` with `linear-gradient`.  Good work.

**Comment by Youssef1313** [comment]
> @Youssef1313
> Is there a way to test this? I'm sure it should be possible...

@carldebilly Not what I can think of :(
I thought these kind of regressions should be caught by screenshot comparing. If not, what do you suggest? It'd be tricky and very hard to test the color of specific pixels as it's not much predictable for a gradient border.

**Comment by jeromelaban** [comment]
It looks like the test is failing on [both android and ios](https://dev.azure.com/uno-platform/Uno%20Platform/_TestManagement/Runs?runId=1561646&_a=resultSummary&resultId=100000). Can you take a look at the screenshots to understand what's missing ?

**Comment by Youssef1313** [comment]
- Android screenshots are correct. Probably a flakiness (it timed out along with an existing test)
- iOS screenshots are incorrect, and it will be handled by https://github.com/unoplatform/uno/issues/6749

@jeromelaban Are you okay with deactivating the test on iOS with a link to #6749?

**Comment by jeromelaban** [comment]
@Youssef1313 yes, let's do this. Add a reference to the issue on the test. Thanks!

**Comment by jeromelaban** [comment]
Looks like the test is not passing on android: https://dev.azure.com/uno-platform/Uno%20Platform/_TestManagement/Runs?runId=1569240&_a=resultSummary&resultId=100000

**Pull request merged**